src/scripts/painter.js

```diff
@@ -0,0 +1,74 @@
+(function() {
+  let ownColour;
+  let originalColour;
+  let reblogColour;
+  let likedColour;
+
+  const paint = async function() {
+    const { timelineObject } = await fakeImport('/src/util/react-props.js');
+
+    [...document.querySelectorAll('[data-id]:not(.xkit_painter_painted)')]
+    .forEach(async postElement => {
+      postElement.classList.add('xkit_painter_painted');
+
+      const post_id = postElement.dataset.id;
+      const { canDelete, liked, rebloggedFromId } = await timelineObject(post_id);
```

Comment:
nitpick, but for regular object destructuring i'm preferring no spaces around the identifiers; i'm using spacing around to specifically denote module syntax

```suggestion
      const {canDelete, liked, rebloggedFromId} = await timelineObject(post_id);
```